### PR TITLE
update casa case detail and cases table to hide unassigned volunteers

### LINKED ISSUE #1233: Remove unassigned volunteers from Volunteers table and Case Detail View
**What is the problem, and what should happen instead?**

Unassigned volunteers still appear in the Case table and Case Detail View. Only volunteers assigned to that case should show up here.

**Reproduction steps**

- While logged in as an Admin, go to Cases to view the cases table
- Click 'Edit' to open a case and assign a few volunteers to it
- On the same page, click 'Unassign Volunteer' to unassign one of them
- Navigate back to Cases table and notice that the case appears to be assigned to all volunteers, including any unassigned
- Click 'Details View' and notice that all volunteers appear here too
- Navigate back to the table, click the unassigned volunteer's name and see that the case is still assigned to them

**Note**: When clicking 'Unassign Cases' from a volunteer's profile page (you get here by clicking a volunteer's name from the table), the Cases table and Case Detail View pages update correctly.

**Screenshots of current behavior, if any**

Screenshot 2020-10-31 at 11 49 10

Screenshot 2020-10-31 at 11 49 00

### PR DESCRIPTION
**Pull request by metamoni**
### What github issue is this PR for, if any?
Resolves #1233

### What changed, and why?
Added a helper method to filter volunteers who are currently assigned to a specific case. Previously, these were displaying the names of all volunteers, including ones who had been unassigned. Applied the helper method into the Cases table and Case Detail View.

### How will this affect user permissions?
- Volunteer permissions: n/a
- Supervisor permissions: n/a
- Admin permissions: n/a

### How is this tested? (please write tests!) 💖💪
Tests added to the `case_casa_helper_spec` file.

### Screenshots please :)
Screenshot 2020-11-04 at 19 21 01

Screenshot 2020-11-04 at 19 21 15

Screenshot 2020-11-04 at 19 21 25

### CODE REVIEW

**Review by compwron — APPROVED** [review]
Yay test! Please fix the merge conflict and then feel free to merge :)

**Comment by compwron** [comment]
I'm going to merge and then cleanup lint etc 😅

File: app/views/casa_cases/show.html.erb [hunk 1]
```diff
@@ -50,7 +50,7 @@
 
     <div>
       <h6><strong>Assigned Volunteers:</strong></h6>
-      <% policy_scope(@casa_case.volunteers).each_with_index do |volunteer, index| %>
+      <% assigned_volunteers(policy_scope(@casa_case)).each_with_index do |volunteer, index| %>
```
Comment (compwron) [inline app/views/casa_cases/show.html.erb:53]: policy has to be on the outside whoops

Comment (Zrrrpy) [reply app/views/casa_cases/show.html.erb:53]: has this been resolved @compwron and was it included in our deploy to prod last week? If yes, I'll move it over to Done so we know it's complete. 😃

Comment (compwron) [reply app/views/casa_cases/show.html.erb:53]: @Zrrrpy yep this is in prod

**Pull request merged**